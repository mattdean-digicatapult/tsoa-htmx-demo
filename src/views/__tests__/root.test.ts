import { describe, expect, it } from 'bun:test'

import RootTemplates from '../root'

describe('RootTemplates', () => {
  describe('Root', () => {
    it('should render root page', async () => {
      const templates = new RootTemplates()
      const rendered = await templates.Root('title', 0)
      expect(rendered).toMatchSnapshot()
    })

    it('should escape html in title', async () => {
      const templates = new RootTemplates()
      const rendered = await templates.Root('<div>Malicious Content</div>', 0)
      expect(rendered).toMatchSnapshot()
    })

    it('should render counter 1 on second load', async () => {
      const templates = new RootTemplates()
      const rendered = await templates.Root('title', 1)
      expect(rendered).toMatchSnapshot()
    })
  })

  describe('Counter', () => {
    it('should render the counter with value 0', async () => {
      const templates = new RootTemplates()
      const rendered = await templates.Counter({ count: 0 })
      expect(rendered).toMatchSnapshot()
    })

    it('should render the counter on second call with value 1', async () => {
      const templates = new RootTemplates()
      const rendered = await templates.Counter({ count: 1 })
      expect(rendered).toMatchSnapshot()
    })
  })

  describe('Button', () => {
    it('should render the enabled button', async () => {
      const templates = new RootTemplates()
      const rendered = await templates.Button({ disabled: false })
      expect(rendered).toMatchSnapshot()
    })

    it('should render the disabled button', async () => {
      const templates = new RootTemplates()
      const rendered = await templates.Button({ disabled: true })
      expect(rendered).toMatchSnapshot()
    })
  })
})
