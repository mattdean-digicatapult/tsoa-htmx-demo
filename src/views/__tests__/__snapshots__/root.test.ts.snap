// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`RootTemplates Root should render root page 1`] = `"<!DOCTYPE html><html lang="en"><head><script src="lib/htmx.org/htmx.min.js"></script><script src="lib/htmx.org/ext/json-enc.js"></script><link rel="icon" type="image/ico" sizes="48x48" href="/public/images/favicon.ico"/><link rel="stylesheet" type="text/css" href="/public/styles/main.css"/><title>title</title></head><body hx-ext="json-enc"><div id="counter" hx-get="/counter" hx-trigger="button-click from:body" hx-swap="outerHTML"><span>0</span><img class="spinner htmx-indicator" src="/public/images/spinner.svg"/></div><button hx-post="/button" hx-swap="outerHTML">Click me!</button></body></html>"`;

exports[`RootTemplates Root should escape html in title 1`] = `"<!DOCTYPE html><html lang="en"><head><script src="lib/htmx.org/htmx.min.js"></script><script src="lib/htmx.org/ext/json-enc.js"></script><link rel="icon" type="image/ico" sizes="48x48" href="/public/images/favicon.ico"/><link rel="stylesheet" type="text/css" href="/public/styles/main.css"/><title>&lt;div&gt;Malicious Content&lt;/div&gt;</title></head><body hx-ext="json-enc"><div id="counter" hx-get="/counter" hx-trigger="button-click from:body" hx-swap="outerHTML"><span>0</span><img class="spinner htmx-indicator" src="/public/images/spinner.svg"/></div><button hx-post="/button" hx-swap="outerHTML">Click me!</button></body></html>"`;

exports[`RootTemplates Root should render counter 1 on second load 1`] = `"<!DOCTYPE html><html lang="en"><head><script src="lib/htmx.org/htmx.min.js"></script><script src="lib/htmx.org/ext/json-enc.js"></script><link rel="icon" type="image/ico" sizes="48x48" href="/public/images/favicon.ico"/><link rel="stylesheet" type="text/css" href="/public/styles/main.css"/><title>title</title></head><body hx-ext="json-enc"><div id="counter" hx-get="/counter" hx-trigger="button-click from:body" hx-swap="outerHTML"><span>1</span><img class="spinner htmx-indicator" src="/public/images/spinner.svg"/></div><button hx-post="/button" hx-swap="outerHTML">Click me!</button></body></html>"`;

exports[`RootTemplates Counter should render the counter with value 0 1`] = `"<div id="counter" hx-get="/counter" hx-trigger="button-click from:body" hx-swap="outerHTML"><span>0</span><img class="spinner htmx-indicator" src="/public/images/spinner.svg"/></div>"`;

exports[`RootTemplates Counter should render the counter on second call with value 1 1`] = `"<div id="counter" hx-get="/counter" hx-trigger="button-click from:body" hx-swap="outerHTML"><span>1</span><img class="spinner htmx-indicator" src="/public/images/spinner.svg"/></div>"`;

exports[`RootTemplates Button should render the enabled button 1`] = `"<button hx-post="/button" hx-swap="outerHTML">Click me!</button>"`;

exports[`RootTemplates Button should render the disabled button 1`] = `"<button disabled hx-trigger="counter-loaded from:body" hx-get="/button" hx-swap="outerHTML">Click me!</button>"`;
